planet/src/main/webapp/places.js

```diff
@@ -148,62 +158,61 @@ function addPlaceDetails(places) {
-                    }
-                }
-            }
-            if (response.result.website != undefined) {
-                placeDetails['Website'] = response.result.website;
-            }
-
-           
-            placeInfo.push(placeDetails);
-        }));
+        // Make request with fields
+        var place = {
+            placeId: place.place_id,
+            fields: ['place_id','name','rating','formatted_phone_number','formatted_address',
+            'opening_hours','photos','url']
+        };
+        // Call Places Details Request
+        let service = new google.maps.places.PlacesService(map);
+        service.getDetails(place, callback); 
     });
-    // Wait for promises to complete before proceeding
-    Promise.all(promises).then(function() {
-        listResults();
-    })
 }
 
-/** Get URL for Place Photo Request  */
-function getPhotoURL(photo_reference) {
-    let baseURL = 'https://maps.googleapis.com/maps/api/place/photo?';
-    let maxWidth = '400';
-    let maxHeight = '200';
-    let photoURL = baseURL + 'maxwidth=' + maxWidth + '&maxheight=' + maxHeight 
-    + '&photoreference=' + photo_reference + '&key=AIzaSyDK36gDoYgOj4AlbCqh1IuaUuTlcpKF0ns'; 
-
-    return photoURL; 
+/** Callback function to handle place details response */
+function callback(place, status) {
+    if (status == google.maps.places.PlacesServiceStatus.OK) {
+        // Add place details to dictionary
+        let placeDetails = {PlaceID: '',Name: '', Rating: '', Address: '', Photo: '',Phone: '',
+        Hours: '', Website: '',Opening: '',Closing: ''}; 
+        placeDetails['PlaceID'] = place.place_id;
+        placeDetails['Name'] = place.name; 
+        // Check if place details exist
+        if (place.rating) {
+            placeDetails['Rating'] = place.rating;    
+        }
+        if (place.formatted_address) {
+            placeDetails['Address'] = place.formatted_address;
+        }
+        if (place.formatted_phone_number) {
+            placeDetails['Phone'] = place.formatted_phone_number;
+        }
+        if (place.opening_hours) {
+            placeDetails['Hours'] = place.opening_hours.weekday_text;
+            // Open and closing time for Monday (MVP purposes)
+            if (place.opening_hours.periods[1]) {
+                placeDetails['Opening'] = place.opening_hours.periods[1].open.time;
+                if (place.opening_hours.periods[1].close) {
+                    placeDetails['Closing'] = place.opening_hours.periods[1].close.time;
+                }
+            }
+        }
+        if (place.photos) {
+            placeDetails['Photo'] = place.photos[0].getUrl({maxWidth:400, maxHeight:200});
+        }
+        if (place.url) {
+            placeDetails['Website'] = place.url;
+        }
+        placeInfo.push(placeDetails);
+    }    
+    
+    // List results when all callbacks are finished
+    count ++ 
+    if (count === places.length) {
```

Comment:
Can you compare placeInfo.length against places.length() instead of maintaining a global |count|? You can then clear the placeInfo list at the end of listResults() (and rename listResults() to listAndClearResults())?
I want to minimize the use of global variables as much as possible since they are hard to keep track of and makes debugging harder.

Reply:
Hmmm okay I've tried testing this and for some reason the placeInfo and places list are not ever the same length. This is because the callback function() line -> if (status === google.maps.places.PlacesServiceStatus.OK) becomes false after 10 places... I'm not sure why this occurs but because of it, only half the returned search results from places list are added to placeInfo

Reply:
I'm going to stick with the count variable for now and try and resolve this error

Reply:
Is it the case that you can only search for 10 places at a time? if so, you should update you client code accordingly to batch or paginate the requests.

Reply:
Also, which case occurs for the results after the first 10? https://developers.google.com/maps/documentation/javascript/reference/places-service#PlacesServiceStatus

Reply:
![image](https://user-images.githubusercontent.com/47041119/88112511-01848000-cb7e-11ea-82cb-a97277a75739.png)

Reply:
It's the OVER_QUERY_LIMIT

Reply:
You can search for 20 places at a time (displayed on the map) but you can only make 10 place details requests (which are the first 10 places)

Reply:
Will that affect the itinerary page as well? (if the user has saved more than 10 places, will they be able to request all place details on the itinerary page?)

Reply:
Please document this issue somewhere and look into how other sites work around this limit. I don't remember the last time I clicked less than 10 places when I saw a map.